flusher.go

```diff
@@ -27,21 +28,48 @@ func Flush(postMetrics [][]DDMetric) {
 		finalMetrics[i].Hostname = Config.Hostname
 	}
 
+	Stats.Gauge("flush.post_metrics_total", float64(totalCount), nil, 1.0)
 	// Check to see if we have anything to do
 	if totalCount == 0 {
 		log.Info("Nothing to flush, skipping.")
 		return
 	}
 
+	// break the metrics into chunks of approximately equal size, such that
+	// each chunk is less than the limit
+	// we compute the chunks using rounding-up integer division
+	workers := ((totalCount - 1) / metricLimit) + 1
+	chunkSize := ((totalCount - 1) / workers) + 1
+	var wg sync.WaitGroup
+	flushStart := time.Now()
+	for i := 0; i < workers; i++ {
+		chunk := finalMetrics[i*chunkSize:]
+		if i < workers-1 {
+			// trim to chunk size unless this is the last one
+			chunk = chunk[:chunkSize]
+		}
+		wg.Add(1)
+		go flushPart(chunk, &wg)
+	}
+	wg.Wait()
+	Stats.TimeInMilliseconds("flush.total_duration_ns", float64(time.Now().Sub(flushStart).Nanoseconds()), nil, 1.0)
+
+	Stats.Count("flush.error_total", 0, nil, 0.1) // make sure this metric is not sparse
```

Comment:
If this comment wasn't here I'd be advocating for 1.0 right now, since n/workers emitting this every 10s is nbd. But since there *is* a comment I'm a bit confused. 😄